### ISSUE TITLE
Deepcopy of control flow with standalone clbit results in invalid circuit

### ISSUE THREAD

**Issue by mtreinish** [Qiskit/qiskit-terra#10409]
### Environment

- **Qiskit Terra version**: 0.24.1 (but also on main)
- **Python version**: 3.11
- **Operating system**:  Linux

### What is happening?

When running a deepcopy of a circuit object using a clbit as a standalone bit as the condition for a control flow operation results in the deepcopied circuit being invalid so that qasm3 or qpy serialization, and visualization will fail because the internal state of the circuit is corrupted and not valid in the copy.

### How can we reproduce the issue?

```python
import copy

from qiskit.circuit import QuantumCircuit, Qubit, Clbit, ClassicalRegister, QuantumRegister, Gate, Parameter

bits = [Qubit(), Qubit(), Clbit()]
qc = QuantumCircuit(bits)
with qc.if_test((qc.clbits[0], 1)):
    qc.x(0)

copy_qc = copy.deepcopy(qc)
print(copy_qc)
```

### What should happen?

It should print the visualization of the circuit. This will stack trace though during the circuit to dag conversion. You can also replace `print(copy_qc)` with something like:

```
import io
from qiskit.qpy import dump

with io.BytesIO() as fd:
    dump(copy_qc, fd)
```
this will also fail on a lookup of the clbit.

### Any suggestions?

I think the special `__deepcopy__` handling in the `qiskit.circuit.Instruction` class is not valid for the `ControlFlowOp` when the `.condition` attribute is a clbit. It might just be simplest to revert to the default deepcopy behavior for `ControlFlowOp`.

**Comment by jakelishman** [comment]
It's the `Bit` instances that are the problem here, really - a deepcopy of an immutable object should return itself. If we make `Bit.__deepcopy__` and `__copy__` return `self`, it should all automatically fix itself.

The handling would fail for a deep-copied regular instruction with a single clbit condition as well, there's just not really any point in our tests where we'd try and and see the failure.

**Comment by mtreinish** [comment]
Yeah that's fair, adding a `Bit.__deepcopy__` to return self would fix this. That's easy enough to add.

**Comment by eendebakpt** [comment]
Perhaps #10385 is related. There also a `deepcopy` of an internal structure results in an exception.
